Documentation/GettingStarted.md

```diff
@@ -1,104 +1,861 @@
+
+* If the `scheduler` is **different** from the thread where `dispose` was being called, then the answer is potentially **yes**.
+
+The reason why that is the case has nothing to do with Rx. You simply could have two processes happening in parallel.
+
+* one is producing elements
+* other on is disposing subscription
+
+A few more examples just to be sure (if you don't know what `observeOn` means, you can return to this part later).
+
+In case you have something like:
+
+```
+let subscription = interval(0.3, scheduler)
+            >- observeOn(MainScheduler.sharedInstance)
+            >- subscribeNext { n in
+                println(n)
+            }
+
+// ....
+
+subscription.dispose() // called from main thread
 
 ```
-a >- b >- c equals c(b(a))
+
+**After `dispose` call returns, no element will be printed. That is a guarantee.**
+
+Also in this case:
+
 ```
+let subscription = interval(0.3, scheduler)
+            >- observeOn(serialScheduler)
+            >- subscribeNext { n in
+                println(n)
+            }
+
+// ...
+
+subscription.dispose() // executing on same `serialScheduler`
+
+```
+
+**After `dispose` call returns, no element will be printed. That is a guarantee.**
+
+**Claims for `Next` event are also valid for `Error` and `Completed` events.**
+
+## Creating your own `Observable` (aka sequence producers)
+
+There is one crucial thing to understand about observables.
+
+**When observable is created it doesn't perform any work just because it has been created. `Observable`s are just a description of how to generate sequences. Nothing more.**
+
+It is true that `Observable` can generate elements in many ways. Some of them cause side effects and some of them register into existing running processes like tapping into mouse events, etc.
 
-Let's first start with some imperative swift code.
-The purpose of example is to bind identifier `c` to a value calculated from `a` and `b` if some condition is satisfied.
+**But if you just call a method that returns an `Observable`, no sequence generation is performed, and there are no side effects. Sequence generation starts when `subscribe` method is called.**
 
-Here is the imperative swift code that calculates the value of `c`:
+E.g. Let's say you have a method with similar prototype:
+
+```
+func searchWikipedia(searchTerm: String) -> Observable<Results> {}
+```
+
+```
+let searchForMe = searchWikipedia("me")
+
+// no requests are performed, no work is being done, no URL requests were fired
+
+let cancel = searchForMe
+  // sequence generation starts now, URL requests are fired
+  >- subscribeNext { results in
+      println(results)
+  }
+
+```
+
+There are a lot of ways how you can create your own `Observable` sequence. Probably the easiest is using `create` function.
+
+Let's code a function that creates a sequence that immediately returns one element upon subscription. That function is called 'just'.
```

Comment:
> Let's code a function that creates a sequence that immediately returns one element upon subscription.

Let's create a function which creates a sequence that returns one element upon subscription.